subsys/bootloader/include/fw_metadata.h

```diff
@@ -24,9 +24,17 @@
 #include <stddef.h>
 #include <toolchain.h>
 #include <assert.h>
+#include <string.h>
+#include <debug.h>
 
 #define MAGIC_LEN_WORDS (CONFIG_SB_MAGIC_LEN / sizeof(u32_t))
 
+#define TYPE_AND_DECL(retval, name, ...) \
+	typedef retval (* name ## _t) (__VA_ARGS__); \
+	retval name (__VA_ARGS__)
+
+struct fw_abi_getter_info;
+
 struct __packed fw_firmware_info {
 	u32_t magic[MAGIC_LEN_WORDS];
```

Comment:
Once we are here, should this be documented?

Reply:
Document the magic word? I could add something like

    /* Magic word to guarantee that this is an instance of fw_firmware_info. */

Reply:
I just wanted to have consistency since the other fields are documented. Perhaps the variable name in itself is sufficient, its up to you.

Reply:
Added docs